data-serving/data-service/test/controllers/preprocessor.test.ts

```diff
@@ -224,4 +282,63 @@ describe('upsert', () => {
             },
         });
     });
+    it('with no existing case does not create a case revision', async () => {
+        const upsertCase = {
+            ...minimalCase,
+            caseReference: {
+                ...minimalCase.caseReference,
+                sourceEntryId: 'case_id',
+            },
+        };
+        const requestBody = {
+            ...upsertCase,
+            curator: { email: 'creator@gmail.com' },
```

Comment:
This poor person's inbox. Think of all the automated testing emails.

Reply:
omg i never thought of that... maybe i'll change in the next PR for EXTRA SAFETY